torch/csrc/distributed/rpc/rref_context.cpp

```diff
@@ -8,8 +8,10 @@ namespace distributed {
 namespace rpc {
 
 RRefContext& RRefContext::getInstance() {
-  static RRefContext context(RpcAgent::getDefaultRpcAgent());
-  return context;
+  // Intentionally leaky singleton; this module may be destructed before
+  // rref.cpp, which may use getInstance() in its module destructor.
```

Comment:
Just to make sure I understand this. `RRefContext` holds a `shared_ptr` to the `RpcAgent`, so the agent will also be alive when `~UserRRef` access it right?

Reply:
Totally agree that we need more changes.

There are two problems here:
 1. the python code isn't shutting down rrefs before the program is about to destruct.
 2. when (1) happens, the program segfaults

This particularly change is all about a 2-line workaround to address (2), rather than the real fix for (1).